scenarios/test_codegen/test/lib_tests/EntityHistory_test.res

```diff
@@ -1025,39 +1067,45 @@ describe("Entity history rollbacks", () => {
     )
   })
 
-  Async.it(
-    "Deep clean prunes history correctly with items in reorg threshold without checking for stale history entities in threshold",
-    async () => {
-      await Db.sql->DbFunctions.EntityHistory.pruneStaleEntityHistory(
-        ~entityName=(module(TestEntity)->Entities.entityModToInternal).name,
-        ~safeChainIdAndBlockNumberArray=[
-          {chainId: 1, blockNumber: 3},
-          {chainId: 2, blockNumber: 2},
-        ],
-        ~shouldDeepClean=false,
-      )
-      let currentHistoryItems = await Db.sql->getAllMockEntityHistory
-
-      let parsedHistoryItems =
-        currentHistoryItems->S.parseJsonOrThrow(TestEntity.entityHistory.schemaRows)
+  Async.it("Prunes history correctly with items in reorg threshold", async () => {
+    let () = await Db.sql->EntityHistory.pruneStaleEntityHistory(
+      ~entityName=(module(TestEntity)->Entities.entityModToInternal).name,
+      ~pgSchema=Env.Db.publicSchema,
+      ~safeReorgBlocks={
+        chainIds: [1, 2],
+        blockNumbers: [3, 2],
+      },
+    )
+    let currentHistoryItems = await Db.sql->getAllMockEntityHistory
 
-      let sort = arr =>
-        arr->Js.Array2.sortInPlaceWith(
-          (a, b) => a.EntityHistory.current.block_number - b.current.block_number,
-        )
+    let parsedHistoryItems =
+      currentHistoryItems->S.parseJsonOrThrow(TestEntity.entityHistory.schemaRows)
 
-      Assert.deepEqual(
-        parsedHistoryItems->sort->stripUndefinedFieldsInPlace,
-        Mocks.historyRows,
-        ~message="Should have deleted the unneeded first items in history",
+    let sort = arr =>
+      arr->Js.Array2.sortInPlaceWith(
+        (a, b) => a.EntityHistory.current.block_number - b.current.block_number,
       )
-    },
-  )
+
+    Assert.deepEqual(
+      parsedHistoryItems->sort->stripUndefinedFieldsInPlace,
+      [
+        Mocks.Chain1.historyRow2,
+        Mocks.Chain2.historyRow2,
+        Mocks.Chain2.historyRow3,
+        Mocks.Chain1.historyRow3,
+      ]->stripUndefinedFieldsInPlace,
```

Comment:
Ok so basically it deep cleans here 👍🏼, doesn't include historyRow1 anymore

Reply:
Yes.